File: front_app/components/models/piyo_shogi_type_info.js
import MemoryRecord from 'js-memory-record'

export class PiyoShogiTypeInfo extends MemoryRecord {
  static get define() {
    return [
      { key: "auto",   name: "自動判別",   message: "基本こちらでよい",                                                              },
      { key: "native", name: "ぴよ将棋",   message: "M1 Mac に「ぴよ将棋」を入れたのに「ぴよ将棋ｗ」が起動して困っている人はこちら", },
      { key: "web",    name: "ぴよ将棋ｗ", message: "なんかしらの理由でアプリ版を入れられない人はこちら",                            },
    ]
  }
}

File: app/models/quick_script/swars/piyo_shogi_config_script/piyo_shogi_type_info.rb
class QuickScript::Swars::PiyoShogiConfigScript
  class PiyoShogiTypeInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: :auto,   name: "スマホだけ",          el_message: "スマホしか使ってない人向け (初期値)",               },
      { key: :native, name: "Mac PC でも表示する", el_message: "ぴよ将棋を最近の Mac にインストールしている人向け", },
    ]
  end
end
